components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -435,84 +433,99 @@ function StarCraft2Player._getMatchupData(player)
 
 	if type(data[1]) == 'table' then
 		for i=1, #data do
-			local plIndex = 1
-			local vsIndex = 2
-			if data[1].match2opponents[2].name == player then
-				plIndex = 2
-				vsIndex = 1
-			end
-			local plOpp = data[1].match2opponents[plIndex]
-			local vsOpp = data[1].match2opponents[vsIndex]
-
-			local prace = CleanRace[plOpp.match2players[1].extradata.faction] or 'r'
-			local orace = CleanRace[vsOpp.match2players[1].extradata.faction] or 'r'
-
-			vs[prace][orace].win = vs[prace][orace].win + (tonumber(plOpp.score or 0) or 0)
-			vs[prace][orace].loss = vs[prace][orace].loss + (tonumber(vsOpp.score or 0) or 0)
-
-			vs['total'][orace].win = vs['total'][orace].win + (tonumber(plOpp.score or 0) or 0)
-			vs['total'][orace].loss = vs['total'][orace].loss + (tonumber(vsOpp.score or 0) or 0)
-
-			vs[prace]['total'].win = vs[prace]['total'].win + (tonumber(plOpp.score or 0) or 0)
-			vs[prace]['total'].loss = vs[prace]['total'].loss + (tonumber(vsOpp.score or 0) or 0)
-
-			vs['total']['total'].win = vs['total']['total'].win + (tonumber(plOpp.score or 0) or 0)
-			vs['total']['total'].loss = vs['total']['total'].loss + (tonumber(vsOpp.score or 0) or 0)
-
+			vs = CustomPlayer._addScoresToVS(vs, data[i].match2opponents, player)
 			years[tonumber(string.sub(data[i].date, 1, 4))] = string.sub(data[i].date, 1, 4)
 		end
 
-		if years[currentYear] ~= nil or years[currentYear - 1] ~= nil or years[currentYear - 2] ~= nil then
+		if years[_CURRENT_YEAR] ~= nil or years[_CURRENT_YEAR - 1] ~= nil or years[_CURRENT_YEAR - 2] ~= nil then
```

Comment:
Not a must, but if wanted, the following would be easier to read, if you switch the two if bodies

```suggestion
		if years[_CURRENT_YEAR] == nil and years[_CURRENT_YEAR - 1] == nil and years[_CURRENT_YEAR - 2] == nil then
```

Reply:
wouldn't `if years[_CURRENT_YEAR] or years[_CURRENT_YEAR - 1] or years[_CURRENT_YEAR - 2] then` be even easier to read?

Reply:
Sure, that works for me